testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -812,39 +821,61 @@ def test_nobonds_warns():
 
 
 class TestSelectionErrors(object):
-    def setUp(self):
-        self.u = make_Universe(('names', 'masses',
-                                'resids', 'resnames', 'resnums'))
-
-    def tearDown(self):
-        del self.u
+    @staticmethod
+    @pytest.fixture()
+    def universe():
+        return make_Universe(('names', 'masses', 'resids', 'resnames', 'resnums'))
 
-    def selection_fail(self, selstr):
-        assert_raises(SelectionError, self.u.select_atoms,
+    def selection_fail(self, selstr, universe):
```

Comment:
now we're not using yield (and having to provide a function to call for each test), we can collapse these functions into the test function itself